### PR TITLE
Feature/139 update node modules

### LINKED ISSUE #139: Update node modules
Update node modules to latest versions where possible. Optionally: check if some modules are still necessary / could be replaced with simpler solutions

### PR DESCRIPTION
**Pull request by ahelmberger**
Resolves #139

The following modules were not updated:

 eslint      ~7.32.0  →   ~8.0.0     (v7 is needed by some plugins, v8 is very new)
 auto-bind    ~4.0.0  →   ~5.0.0     (v5 is ESM only)
 react      ~16.14.0  →  ~17.0.2     (v16 is needed by some components as peer)
 react-dom  ~16.14.0  →  ~17.0.2     (see above)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,79 +21,76 @@
...
+    "@babel/register": "~7.15.3",
+    "@fortawesome/fontawesome-free": "~5.15.4",
     "@unly/iso3166-1": "~1.0.2",
     "abcjs": "~5.12.0",
     "accept-language-parser": "~1.5.0",
     "acho": "~4.0.6",
-    "antd": "~4.9.4",
+    "antd": "~4.16.13",
     "any-base": "~1.1.0",
-    "array-shuffle": "~2.0.0",
```
Comment (adrianaluchian) [inline package.json:36]: 👍

File: package.json [hunk 2]
```diff
@@ -108,35 +105,34 @@
...
-    "yaml": "~1.10.0"
+    "yaml": "~1.10.2"
   },
   "devDependencies": {
-    "acorn": "~8.0.4",
-    "babel-jest": "~26.6.3",
+    "acorn": "~8.5.0",
+    "babel-jest": "~27.2.5",
     "babel-loader": "~8.2.2",
-    "delay": "~4.4.0",
```
Comment (adrianaluchian) [inline package.json:117]: 👍

File: src/utils/shuffle-array.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+export default function shuffleArray(array) {
```
Comment (adrianaluchian) [inline src/utils/shuffle-array.js:1]: Let's'a add tests for this helper 🙏 

Also, I'd suggest reducing the utils modules by grouping a bit. We could do the classic group by data type and have a `array-utils.js` (I see there's a nice `string-utils.js` already).

File: src/utils/array-utils.spec.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    });
+    it('does not mutate the original array if the item was included in the original array', () => {
+      const input = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
+      const result = ensureIsExcluded(input, 4);
+      expect(result).not.toBe(input);
+      expect(input).toEqual([0, 1, 2, 3, 4, 5, 6, 7, 8, 9]);
+    });
+  });
+
+  describe('shuffleItems', () => {
```
Comment (adrianaluchian) [inline src/utils/array-utils.spec.js:86]: We could even add a nice check if it is shuffled :)

For example we could do 
```
 expect(input.join()).not.toEqual(result.join());
```
And if you're considering maybe likelihood of a shuffle with exact output as original, we could shuffle twice and do something like 
```
expect(input.join() !== result1.join() || input.join() !== result2.join()).toBe(true);
```

What do you think?

File: src/utils/array-utils.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  const lastIndex = items.length - 1;
+  if (index1 < 0 || index2 < 0 || index1 > lastIndex || index2 > lastIndex || index1 === index2) {
+    return items;
+  }
+
+  const result = items.slice();
+
+  const item1 = result[index1];
+  const item2 = result[index2];
+  result[index1] = item2;
```
Comment (laurentiu-ilici) [inline src/utils/array-utils.js:11]: or: 
result[index1] = items[index2]
result[index2] = items[index1]
No need for the extra variables

File: src/utils/array-utils.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+  return items.filter((t, i) => i !== index);
+}
+
+export function ensureIsIncluded(items, item) {
+  return items.includes(item) ? items : [...items, item];
+}
+
+export function ensureIsExcluded(items, item) {
+  return items.includes(item) ? items.filter(i => i !== item) : items;
```
Comment (laurentiu-ilici) [inline src/utils/array-utils.js:30]: There's no need for the include ternary, the filter will exclude the item anyway.

File: src/utils/array-utils.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+export function shuffleItems(items) {
+  const result = items.slice();
+
+  for (let index = result.length - 1; index > 0; index -= 1) {
+    const newIndex = Math.floor(Math.random() * (index + 1));
+    [result[index], result[newIndex]] = [result[newIndex], result[index]];
```
Comment (laurentiu-ilici) [inline src/utils/array-utils.js:38]: Fancy

File: src/utils/array-utils.spec.js (hunk 4 above)
Comment (laurentiu-ilici) [reply src/utils/array-utils.spec.js:86]: The second variant might work in a lot more cases, but it's in theory an unstable test.

File: src/stores/order-store-base.js [hunk 8]
```diff
@@ -9,7 +9,7 @@ class OrderStoreBase extends StoreBase {
   async getNextOrder() {
     const query = { _id: this.orderKey };
     const update = { $inc: { seq: 1 } };
-    const options = { upsert: true, returnOriginal: false };
+    const options = { upsert: true, returnDocument: 'after' };
```
Comment (laurentiu-ilici) [inline src/stores/order-store-base.js:12]: I guess this is a breaking change in the driver...

Comment (ahelmberger) [reply src/stores/order-store-base.js:12]: No, it's just a different way to configure the same thing. The behavior won't change.

**Pull request merged**